core-backend/sandbox/src/env.rs

```diff
@@ -73,7 +75,6 @@ pub(crate) struct Runtime<E: Ext> {
     pub ext: ExtCarrier<E>,
     pub memory: MemoryWrap,
     pub trap: Option<FuncError<E::Error>>,
```

Comment:
how do you feel to keep here not the Option with error, but the "actual TerminationReason"? It starts with `Success` and changes to other variants in appropriate cases? could we do something like that?